numpy/_core/src/umath/ufunc_object.c

```diff
@@ -984,6 +984,10 @@ try_trivial_single_output_loop(PyArrayMethod_Context *context,
         res = -1;
     }
```

Comment:
Here is the "doesn't correctly return -1 because it's an old style loop" logic.  I.e. it is dealt with explicitly before the function is called. (It is also unrelated to the "needs API" flag since it is valid and typical to grab the GIL to set an error.)